src/metabase/driver.clj

```diff
@@ -304,7 +304,7 @@
         java.math.BigInteger            :BigIntegerField
         java.sql.Date                   :DateField
         java.sql.Timestamp              :DateTimeField
-        java.util.Date                  :DateField
+        java.util.Date                  :DateTimeField
```

Comment:
It turns out the frontend is not properly accounting for this, but all the same we should be as consistent as we can and that should be fixed up as detailed in #1929 

In any case, the `java.util.Date` class allows for specifying a time component down to milliseconds, so it makes sense to me for this to be treated as a `:DateTimeField`.  On the other hand the `java.sql.Date` class is specifically a subclass of java.util.Date that removes/ignores the time component in order to accurately represent a `Date` column in databases, so that one is left mapping to `:DateField`